### DISCUSSION TITLE
Building on Debian 9 fails on mysqlclient-sys

### DISCUSSION THREAD

**Discussion by Brawl345** [dani-garcia/vaultwarden#1603] [Q&A]
Hello, I'm trying to manually build bitwarden_rs on Debian 9 Stretch because I don't want to use Docker and extracting the image doesn't work for me because I need MySQL...

But I'm getting this error when running `cargo build --features mysql --release`:

```
error: empty search path given via `-L`

error: could not compile `mysqlclient-sys`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

I've installed all the dependencies listed here: https://github.com/dani-garcia/bitwarden_rs/wiki/Building-binary

Does it even work on Debian 9...? Has someone tested it on 10? I would be very thankful for any help!

**Comment by BlackDex** [comment]
I think you may also need pkg-config for it to work.
But are you sure you have installed both the mariadb packages?

**Reply by Brawl345** [reply]
Thanks for the help.
I've installed pkg-config, libmariadb-dev-compat and libmariadb-dev already :/

**Reply by jjlin** [reply]
The Docker build uses the [`rust`](https://hub.docker.com/_/rust) image, which is based on Debian buster. If you're running Debian buster, you can just extract the binary from the `bitwardenrs/server:latest` image. Otherwise you can follow the Dockerfiles to see what dependencies you're missing:

* https://github.com/dani-garcia/bitwarden_rs/blob/master/docker/amd64/Dockerfile
* https://github.com/rust-lang/docker-rust/blob/master/1.51.0/buster/Dockerfile
* https://github.com/docker-library/buildpack-deps/blob/master/debian/buster/Dockerfile

**Reply by Brawl345** [reply]
It's Debian Stretch sadly, so extracting the build doesn't help. I also installed all of the dependencies from the Dockerfiles but I still get the same error :/
